### PR TITLE
[Feature/#270] Support vm.getBlockNumber cheatcode

### LINKED ISSUE #270: vm.getBlockNumber not supported
**Describe the bug**
When using vm cheatcode from foundry we get the error : `WARNING:Halmos:Encountered Unsupported cheat code: calldata = 0x42cbb15c` 
Can be bypass using block.number directly on the code but could be a nice addition to Halmos

**To Reproduce**
Use vm.getBlockNumber() in your test file

**Environment:**
 - OS: macOs
 - Python version: 3.12.2
 - Halmos and other dependency versions: halmos 0.1.11

**Additional context**
cheatcode not supported

### PR DESCRIPTION
**Pull request by pillip**
## Change Type

- [ ] Bug fix (Not Breaking-change. Solve issue.)
- [x] New feature (Not Breaking-change. Add new feature.)
- [ ] Breaking change (Bug fix or New featrure that cause Breaking change.)
- [ ] Code Convention (Edit Code Convention.)
- [ ] Refactoring (Refactor Code without feature change.)

## Description
This PR introduces supporting vm.getBlockNumber() cheat code. Key changes include the `getBlockNumber()` cheat code function to retrieve the current block number and adding solidity test contract to verify the functionality of `getBlockNumber`.

## Related Issues
This PR resolves #270

## Changes

1. Cheat Code Implementation
    * Added handling for `getBlockNumber()` in the cheat code logic.
    * Added the solidity test contract

2. Update submodules

## Tests
```
# at the ~~~~/halmos/example/simple dirctory
halmos -vvvv --contract BlockNumberTest
```

### CODE REVIEW

**Review by daejunpark — COMMENTED** [review]
thanks for this!

if it is necessary to update the forge-std submodule, this can be merged only after #323. 

also, to make the scope of changes clearer, i'd suggest not updating irrelevant submodules in this pr.

additional comments below.

File: examples/simple/remappings.txt [hunk 1]
```diff
@@ -1 +1,4 @@
 multicaller/=../../tests/lib/multicaller/src/
+openzeppelin/=../../tests/lib/openzeppelin-contracts/contracts/
+ds-test/=../../tests/lib/forge-std/lib/ds-test/src/
+forge-std/=../../tests/lib/forge-std/src/
```
[the remark this answers is not in the document]
Comment (pillip) [reply examples/simple/remappings.txt]: @daejunpark 
This comment is resolved at f78d87c

**Comment by pillip** [comment]
@daejunpark 
I reverted updates of irrelevant submodules at 57bf7d3 
Thank you for feedback!

**Comment by daejunpark** [comment]
thanks! we will merge this after #323.

**Review by daejunpark — APPROVED** [review]

**Pull request merged**